addon/index.js

```diff
@@ -0,0 +1,304 @@
+
+function _fmt(str, formats) {
+  let cachedFormats = formats;
+
+  if (!Array.isArray(cachedFormats) || arguments.length > 2) {
+    cachedFormats = new Array(arguments.length - 1);
+
+    for (let i = 1; i < arguments.length; i++) {
+      cachedFormats[i - 1] = arguments[i];
+    }
+  }
+
+  // first, replace any ORDERED replacements.
+  let idx  = 0; // the current index for non-numerical replacements
+  return str.replace(/%@([0-9]+)?/g, (s, argIndex) => {
+    argIndex = (argIndex) ? parseInt(argIndex, 10) - 1 : idx++;
+    s = cachedFormats[argIndex];
+    return (s === null) ? '(null)' : (s === undefined) ? '' : inspect(s);
+  });
+}
+
+/**
+  Apply formatting options to the string. This will look for occurrences
+  of "%@" in your string and substitute them with the arguments you pass into
+  this method. If you want to control the specific order of replacement,
+  you can add a number after the key as well to indicate which argument
+  you want to insert.
+
+  Ordered insertions are most useful when building loc strings where values
+  you need to insert may appear in different orders.
+
+  ```javascript
+  "Hello %@ %@".fmt('John', 'Doe');     // "Hello John Doe"
+  "Hello %@2, %@1".fmt('John', 'Doe');  // "Hello Doe, John"
+  ```
+
+  @method fmt
+  @param {String} str The string to format
+  @param {Array} formats An array of parameters to interpolate into string.
+  @return {String} formatted string
+  @public
+  @deprecated Use ES6 template strings instead: http://babeljs.io/docs/learn-es2015/#template-strings
+*/
+export function fmt() {
+  deprecate(
+    'fmt is deprecated, use ES6 template strings instead.',
+    false,
+    { id: 'ember-string-utils.fmt', until: '3.0.0', url: 'http://babeljs.io/docs/learn-es2015/#template-strings' }
+  );
+  return _fmt(...arguments);
+}
+
+/**
+  Formats the passed string, but first looks up the string in the localized
+  strings hash. This is a convenient way to localize text. See
+  `Ember.String.fmt()` for more information on formatting.
+
+  Note that it is traditional but not required to prefix localized string
+  keys with an underscore or other character so you can easily identify
+  localized strings.
+
+  ```javascript
+  Ember.STRINGS = {
+    '_Hello World': 'Bonjour le monde',
+    '_Hello %@ %@': 'Bonjour %@ %@'
+  };
+
+  Ember.String.loc("_Hello World");  // 'Bonjour le monde';
+  Ember.String.loc("_Hello %@ %@", ["John", "Smith"]);  // "Bonjour John Smith";
+  ```
+
+  @method loc
+  @param {String} str The string to format
+  @param {Array} formats Optional array of parameters to interpolate into string.
+  @return {String} formatted string
+  @public
+*/
+export function loc(str, formats) {
+  deprecate(
+    'loc is deprecated, use an internationalization or localization addon instead.',
+    false,
+    { id: 'ember-string-utils.loc', until: '3.5.0', url: 'http://babeljs.io/docs/learn-es2015/#template-strings' }
```

Comment:
Should this be a link to ember-intl or ember-i18n or some other page about the topic?